modules/queue/manager.go

```diff
@@ -198,17 +198,15 @@ func (m *Manager) FlushAll(baseCtx context.Context, timeout time.Duration) error
 					wg.Done()
 				}(mq)
 			} else {
-				log.Debug("Queue: %s is non-empty but is not flushable - adding 100 millisecond wait", mq.Name)
-				go func() {
-					<-time.After(100 * time.Millisecond)
-					wg.Done()
-				}()
+				log.Debug("Queue: %s is non-empty but is not flushable", mq.Name)
+				wg.Done()
 			}
-
 		}
 		if allEmpty {
+			log.Debug("All queues are empty")
 			break
 		}
+		<-time.After(100 * time.Millisecond)
```

Comment:
Add a time.C in line 174 is better.

Reply:
The point of the delay is to make every loop take at least 100ms. Moving the timer up to 174 would delay the start of the flusher for 100ms which is not quite the same thing. 

Now we could move the cancellation test in 174 below the wg.Wait but that doesn't really account for the time taken to get the managed queues.

I think the correct solution is therefore to retest against the ctx before the wait. That will allow the wait groups to tidy up during the abort from the cancelled context without further delay.